dace/sdfg/utils.py

```diff
@@ -1030,6 +1031,114 @@ def get_next_nonempty_states(sdfg: SDFG, state: SDFGState) -> Set[SDFGState]:
+        else:
+            yield 0
+    else:
+        yield 0
+
+
+def modify_distributed_subset(subset: Union[sbs.Subset, list, tuple],
+                              change: int):
+    """
+    Applies changes to magic indices from a subset if array is a HBM-array, otherwise
+    returns subset. subset is deepcopied before any modification to it is done.
+    :param change: magic index is set to this value, unless it's -1 in which case
+        the magic index is completly removed
+    :param force: Modify the first index even if this is not a HBM array
+    """
+    cps = copy.deepcopy(subset)
+    if isinstance(subset, sbs.Subset):
+        if change == -1:
+            cps.pop([0])
+        else:
+            cps[0] = (change, change, 1)
+    elif isinstance(subset, list) or isinstance(subset, tuple):
+        if isinstance(subset, tuple):
+            cps = list(cps)
+        if change == -1:
+            cps.pop(0)
+        else:
+            cps[0] = change
+        if isinstance(subset, tuple):
+            cps = tuple(cps)
+    else:
+        raise ValueError(
+            "unsupported type passed to modify_distributed_subset")
+    
+    return cps
+
+
+def get_multibank_ranges_from_subset(subset: Union[sbs.Subset, str],
+                                     sdfg: SDFG) -> Tuple[int, int]:
+    """
+    Returns the upper and lower end of the accessed HBM-range, evaluated using the
+    constants on the SDFG.
+    :param assumeSingle: Throw error if more then a single bank is accessed
+    :param codegenlocation: An optional string describing where in the SDFG 
+        codegen the evaluation takes place. If set, this function will throw 
+        CodegenErrors instead of ValueErrors.
+    :returns: (low, high) where low = the lowest accessed bank and high the 
+        highest accessed bank + 1.
+    """
+    if isinstance(subset, str):
+        subset = sbs.Range.from_string(subset)
+    low, high, stride = subset[0]
+    if stride != 1:
+        raise NotImplementedError(f"Strided HBM subsets not supported.")
+    try:
+        low = int(symbolic.resolve_symbol_to_constant(low, sdfg))
+        high = int(symbolic.resolve_symbol_to_constant(high, sdfg))
+    except:
+        raise ValueError(
+            f"Only constant evaluatable indices allowed for HBM-memlets on the bank index."
+        )
+    return (low, high + 1)
+
+
+def parse_location_bank(array: dt.Array) -> Tuple[str, str]:
+    if "bank" in array.location:
+        val: str = array.location["bank"]
+        split = val.split(".")
+        if (len(split) != 2):
+            raise ValueError(
+                f"Failed to parse {val} as value for location['bank']")
+        split[0] = split[0].upper()
+
+        if (split[0] == "DDR" or split[0] == "HBM"):
+            return (split[0], split[1])
+        else:
+            raise ValueError(
+                f"{split[0]} is an invalid bank type for location['bank']. Supported are HBM and DDR."
+            )
```

Comment:
In the error message, write specifically what format is expected, e.g., `f"Failed to parse memory bank specifier \"{val}\". Expected format <type>.<id> (e.g., ddr.3 or hbm.0:4)."`